web/modules/custom/routingsystem/routingsystem.module

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/**
+ * @file
+ * This file revokes the permission of editor role using hook_entity_access .
+ */
+
+use Drupal\user\Entity\Role;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+
+/**
+ * Implements hook_entity_access().
+ */
+function routingsystem_entity_access($entity, $operation, $account) {
+  $routeInfo = \Drupal::routeMatch()->getRouteName();
+  if ($routeInfo === 'routingsystem.route') {
+    if ($account->hasRole('editor')) {
+      $role = Role::load('editor');
+      if ($role) {
+        $role->revokePermission('routingsystem.permission');
+        $role->save();
+        $response = new RedirectResponse(\Drupal::request()->getRequestUri());
+        $response->send();
```

Comment:
You are trying to do a one time activity here.
But this hook will get called for all entities in the system when they are being loaded.
So the hook you have chosen is not the optimal hook for this.